src/frontend/sharedComponents/MediaScrollView/AudioThumbnail.tsx

```diff
@@ -36,25 +33,24 @@ export const AudioThumbnail: FC<AudioThumbnailProps> = ({
   }
 
   const handlePress = async () => {
-    setLoading(true);
-    let uri: string | undefined;
-    const isSavedUri = isAudioAttachment(audioAttachment);
-
     if (isUnsavedAudio(audioAttachment)) {
-      uri = audioAttachment.uri;
-    } else {
-      uri = await projectApi.$blobs.getUrl({
-        driveId: audioAttachment.driveDiscoveryId,
-        name: audioAttachment.name,
-        type: audioAttachment.type,
-        variant: 'original',
+      navigation.navigate('AudioPlaybackUnsavedPreview', {
+        uri: audioAttachment.uri,
       });
+      return;
     }
+
+    setLoading(true);
+    const uri = await projectApi.$blobs.getUrl({
+      driveId: audioAttachment.driveDiscoveryId,
+      name: audioAttachment.name,
+      type: audioAttachment.type,
+      variant: 'original',
+    });
     setLoading(false);
-    navigation.navigate('Audio', {
-      isEditing: currentRoute?.name === 'ObservationEdit',
+    navigation.navigate('AudioPlaybackSaved', {
       uri,
-      isSavedUri,
+      canDelete: currentRoute?.name === 'ObservationEdit',
```

Comment:
i don't think shared components should have checks like this, as it results in an implicit dependency on the navigation context which makes them less reusable. ideally this is derived based on a explicitly passed prop into this component (e.g. `canDelete?: boolean`) + some checks in this component related to the URI in question (similar to before when `isSavedUri` is calculated).

May potentially require a little rewiring because this is used pretty deeply within a couple of shared components.

Reply:
I agree, but you are right that it is too nested to do anything right now. This component will eventually be refactored (with the the update of the observation hook). I will add a to do note to that when component is refactored